plugins/inputs/mesos/mesos_test.go

```
@@ -659,100 +610,140 @@ func startTestServer(t *testing.T) *httptest.Server {
-			CACertificatePath: pki.CACertPath(),
-			IAMConfigPath:     tmpFile,
-			UserAgent:         testUserAgent,
-		},
-	}
-
-	client, err := m.createHttpClient()
-	require.NoError(err)
-	require.NotNil(client)
-	require.NotNil(client.Transport)
+	for _, tc := range TEST_CASES {
+		t.Run(tc.fixture, func(t *testing.T) {
+			if tc.iam {
+				server := startTestServer(t)
+				defer server.Close()
 
-	// test that all requests on client send an Authorization header with token and configured User-Agent header
-	tokenTestServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		auth := r.Header.Get("Authorization")
-		if auth != "token="+testToken {
-			t.Fatalf("Expected request header: `Authorization: token=%s`. Got: %s", testToken, auth)
-		}
-		userAgent := r.Header.Get("User-Agent")
-		if userAgent != testUserAgent {
-			t.Fatalf("Expected request header: `User-Agent: %s`. Got: %s", testUserAgent, userAgent)
-		}
-	}))
-	defer tokenTestServer.Close()
+				// create temp json file using mocked server.URL for loginEndpoint
+				modifiedServiceAcctString := strings.Replace(pki.ReadIAMAccount(), "http://127.0.0.1:8101", server.URL, 1)
+				require.Nil(ioutil.WriteFile(tmpServiceAcctFile, []byte(modifiedServiceAcctString), 0644))
+				defer os.Remove(tmpServiceAcctFile)
+			}
 
-	c := http.Client{
-		Transport: client.Transport,
-	}
+			m := Mesos{
+				ClientConfig: tc.tlsConfig,
+				DCOSConfig:   tc.dcosConfig,
+			}
+			client, err := m.createHttpClient()
+			require.Equal(tc.expClientErr, err)
+			if err == nil {
+				require.NotNil(client)
+				require.NotNil(client.Transport)
+			}
 
-	resp, err := c.Get(tokenTestServer.URL)
-	if err != nil {
-		t.Fatal(err)
+			if tc.iam {
+				// test that all requests on client send an Authorization header with token
+				tokenTestServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					if err := tc.checkFunc(r.Header.Get(tc.header)); err != nil {
```

Looks like we only verify headers if `iam` is `true`, which is surprising. I could see this causing confusion: I might define a test case that has a header name and value, but forget to set `iam` (either by accident or because I assume I don't need it for a header unrelated to IAM). The test would pass while silently skipping the header assertion, and I probably wouldn't think to investigate further. How about we get rid of the `iam` field, and instead check `if tc.header != ""` to decide whether we need to inspect request headers?